### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by erich**
https://trello.com/c/9SnEkw5o/17-authentication

### CODE REVIEW

File: lib/reddex_web/plugs/auth.ex [hunk 1]
```diff
@@ -0,0 +1,21 @@
+defmodule ReddexWeb.Plugs.Auth do
+  @moduledoc "Check if current_user is present in the session"
+
+  import Plug.Conn
+
+  def init(opts), do: opts
+
+  def call(conn, _) do
+    # TODO: Not sure about this
```
Comment (Math3v) [inline lib/reddex_web/plugs/auth.ex]: TODO found

File: lib/reddex_web/plugs/auth.ex [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    # maybe be more explicit in controllers with actions
+    if get_session(conn, :current_user) || conn.request_path == "/sign_in" do
+      conn
+    else
+      conn |> redirect_to_login()
```
Comment (Math3v) [inline lib/reddex_web/plugs/auth.ex]: Use a function call when a pipeline is only one function long

**Comment by Math3v** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 2 possible new issues (including those that may have been commented here).

You can see more details about this review at https://ebertapp.io/github/PrimeHammer/reddex/pulls/14.

**Comment by Math3v** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 1 possible new issue (including those that may have been commented here).

You can see more details about this review at https://ebertapp.io/github/PrimeHammer/reddex/pulls/14.

File: lib/reddex_web/plugs/auth.ex [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  @moduledoc "Check if current_user is present in the session"
+
+  import Plug.Conn
+
+  def init(opts), do: opts
+
+  def call(conn, _) do
+    # TODO: Not sure about this
+    # maybe be more explicit in controllers with actions
+    if get_session(conn, :current_user) || conn.request_path == "/sign_in" do
```
Comment (Math3v) [inline lib/reddex_web/plugs/auth.ex]: I'd leave it like this for now and refactor later, when more "public" pages are at place.

File: lib/reddex_web/plugs/auth.ex [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+      conn
+    else
+      redirect_to_login(conn)
+    end
+  end
+
+  defp redirect_to_login(conn) do
+    conn |> Phoenix.Controller.redirect(to: "/sign_in") |> halt()
```
Comment (Math3v) [inline lib/reddex_web/plugs/auth.ex]:
```
conn
|> Phoenix.Controller.redirect(to: "/sign_in")
|> halt()
``` ?

File: lib/reddex_web/plugs/auth.ex (hunk 3 above)
Comment (erich) [reply lib/reddex_web/plugs/auth.ex]: ok

File: lib/reddex_web/plugs/auth.ex (hunk 4 above)
Comment (erich) [reply lib/reddex_web/plugs/auth.ex]: i'm not sure, i can change but function name and length should be clear enough from my point of view

**Pull request merged**